src/lib.rs

```diff
@@ -151,6 +151,10 @@ pub struct Metadata {
     pub packages: Vec<Package>,
     /// A list of all workspace members
     pub workspace_members: Vec<PackageId>,
+    /// A list of all workspace members
+    ///
+    /// This is always `None` if running with a version of Cargo older than 1.71.
+    pub workspace_default_members: Option<Vec<PackageId>>,
```

Comment:
Is this `Vec` ever empty? If not, then we could avoid the `Option` and use an empty vec as the value signifying that cargo is too old

Reply:
ok, in that case create a `WorkspaceDefaultMembers` struct that has a private field storing the `Option<Vec<PackageId>>`.

We then have a few options:

* implement `Deref` for it, panicking if it is none, otherwise returning `&[PackageId]`
* keep your `workspace_default_packages` method
* use the same method, but don't return an `Option`, by populating it with the right values (just grab the `workspace_members` list?)
* do some shenanigans during deserialization of `Metadata`, copying `workspace_members` if `workspace_default_members` does not exist in the serialized data.

opinions?

Reply:
Oh. I wasn't aware of this issue. Thanks.

I guess let's go with one of the panicking versions then?